### PR TITLE
Replace Uyuni configuration formula with an API call

### PR DESCRIPTION
**Pull request by nadvornik**
## What does this PR change?

This PR implements API call equivalent to Uyuni configuration formula.

All the logic is handled in java,  except for salt minion id matching, which is implemented
via salt runner call.

## GUI diff

No difference.

- [ ] **DONE**

## Documentation
- TBD

- [ ] **DONE**

## Test coverage
- Unit tests were added

- [x] **DONE**

## Links

https://github.com/SUSE/spacewalk/issues/22499

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/admin/configuration/AdminConfigurationHandler.java [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+                details.put("first_name", firstName);
+                details.put("last_name", lastName);
+                details.put("email", email);
+                userHandler.setDetails(loggedInUser, name, details);
+            }
+        }
+        catch (Exception e) {
+            userHandler.create(loggedInUser, name, password, firstName, lastName, email, usePamAuth ? 1 : 0);
+            LOG.debug("User {} created", name);
+        }
```
Comment (cbosdo) [inline java/code/src/com/redhat/rhn/frontend/xmlrpc/admin/configuration/AdminConfigurationHandler.java:214]: Could be factorized with the same code in `updateAdminUser()`

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/admin/configuration/AdminConfigurationHandler.java [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+        String key = String.format("%d-%s", orgId, name);
+        try {
+            ActivationKey details = activationKeyHandler.getDetails(loggedInUser, key);
+            LOG.debug("Activation Key found: {} {}", name, details.getNote());
+
+            if (details.getNote() != description ||
+                details.getBaseChannel().getLabel() != baseChannel ||
+                details.getUsageLimit().intValue() != usageLimit ||
+                details.getContactMethod().getName() != contactMethod) {
+                LOG.debug("Updating Activation Key details: {}", name);
```
Comment (cbosdo) [inline java/code/src/com/redhat/rhn/frontend/xmlrpc/admin/configuration/AdminConfigurationHandler.java:316]: May be you could add a blank line at the start of the block to better visualize the end of conditions.

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/admin/configuration/AdminConfigurationHandler.java [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+        activationKeyHandler.setConfigChannels(loggedInUser, Arrays.asList(key), configurationChannels);
+    }
+
+    /**
+     * Configure server.
+     * @param loggedInUser the current user
+     * @param content the Uyuni configuration formula data
+     * @return 1 on success
+     *
+     * @apidoc.doc Configure server.
```
Comment (cbosdo) [inline java/code/src/com/redhat/rhn/frontend/xmlrpc/admin/configuration/AdminConfigurationHandler.java:405]:
```suggestion
     * @apidoc.doc Configure the server.
```

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/admin/configuration/test/AdminConfigurationHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+
+        config = config.replace("MANAGEABLE_CHANNEL", "");
+        config = config.replace("SUBSCRIBABLE_CHANNEL", "");
+        config = config.replace("CONFIGURATION_CHANNEL", "");
+
+//        HibernateFactory.commitTransaction();
+//        HibernateFactory.closeSession();
+
+
+//        Thread.sleep(4000000);
```
Comment (cbosdo) [inline java/code/src/com/redhat/rhn/frontend/xmlrpc/admin/configuration/test/AdminConfigurationHandlerTest.java:177]:
```suggestion
```

**Review by cbosdo — APPROVED** [review]
Just a few cosmetic remarks. Thanks a lot for the work!

**Review by ycedres — APPROVED** [review]

**Pull request merged**